src/test/operators/table_scan_sorted_test.cpp

```diff
@@ -0,0 +1,175 @@
+
+    const bool ascending = _order_by == OrderByMode::Ascending || _order_by == OrderByMode::AscendingNullsLast;
+    const bool nulls_first = _order_by == OrderByMode::Ascending || _order_by == OrderByMode::Descending;
+
+    if (!ascending) {
+      std::reverse(_expected.begin(), _expected.end());
+    }
+
+    _table_column_definitions.emplace_back("a", _data_type, nullable);
+
+    auto table = Table::create_dummy_table(_table_column_definitions);
+
+    if (nullable && nulls_first) {
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+    }
+
+    const int table_size = 10;
+    for (int i = 0; i < table_size; i++) {
+      if (_data_type == DataType::Int) {
+        table->append({ascending ? i : table_size - i - 1});
+        if (use_reference_segment) {
+          table->append({ascending ? i : table_size - i - 1});
+        }
+      } else if (_data_type == DataType::String) {
+        table->append({pmr_string{std::to_string(ascending ? i : table_size - i - 1)}});
+        if (use_reference_segment) {
+          table->append({pmr_string{std::to_string(ascending ? i : table_size - i - 1)}});
+        }
+      } else {
+        Fail("Unsupported DataType");
+      }
+    }
+
+    if (nullable && !nulls_first) {
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+      table->append({NULL_VALUE});
+    }
+
+    ChunkEncoder::encode_all_chunks(table, SegmentEncodingSpec(_encoding_type));
+
+    const auto ordered_by = std::make_pair(ColumnID(0), _order_by);
+    table->get_chunk(ChunkID(0))->set_ordered_by(ordered_by);
+
+    if (use_reference_segment) {
+      auto pos_list = std::make_shared<PosList>();
+
+      if (nullable && nulls_first) {
+        pos_list->emplace_back(INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET);
+        pos_list->emplace_back(INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET);
+      }
+
+      for (auto i = 0; i < 2 * table_size; ++i) {
+        if (i % 2 == 0) {
+          pos_list->emplace_back(ChunkID(0), i + (nullable && nulls_first ? 3 : 0));
+        }
+      }
+
+      if (nullable && !nulls_first) {
+        pos_list->emplace_back(INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET);
+        pos_list->emplace_back(INVALID_CHUNK_ID, INVALID_CHUNK_OFFSET);
+      }
+
+      const auto reference_segment = std::make_shared<ReferenceSegment>(table, ColumnID(0), pos_list);
+
+      auto reference_table = std::make_shared<Table>(_table_column_definitions, TableType::References);
+      reference_table->append_chunk({reference_segment});
+      reference_table->get_chunk(ChunkID(0))->set_ordered_by(ordered_by);
+
+      _table_wrapper = std::make_shared<TableWrapper>(std::move(reference_table));
+    } else {
+      _table_wrapper = std::make_shared<TableWrapper>(std::move(table));
+    }
+
+    _table_wrapper->execute();
+  }
+
+  void ASSERT_COLUMN_SORTED_EQ(std::shared_ptr<const Table> table) {
```

Comment:
All cap names should be reserved to constexprs/macros. (I know there are probably a few other cases in the code base violating this...)